src/main/java/org/medicmobile/webapp/mobile/RequestLocationPermissionActivity.java

```diff
@@ -34,10 +36,20 @@ public class RequestLocationPermissionActivity extends FragmentActivity {
 
 	private final ActivityResultLauncher<String[]> requestPermissionLauncher =
 		registerForActivityResult(new ActivityResultContracts.RequestMultiplePermissions(), grantedMap -> {
-			boolean allGranted = grantedMap.size() > 0 && !grantedMap.containsValue(false);
-
-			if (allGranted) {
-				trace(this, "RequestLocationPermissionActivity :: User allowed location permission.");
+			boolean fineLocationGranted = Optional
+				.ofNullable(grantedMap.get(ACCESS_FINE_LOCATION))
+				.orElse(false);
+			boolean coarseLocationGranted = Optional
+				.ofNullable(grantedMap.get(ACCESS_COARSE_LOCATION))
+				.orElse(false);
```

Comment:
```suggestion
			grantedMap.getOrDefault(ACCESS_FINE_LOCATION, false);
			grantedMap.getOrDefault(ACCESS_COARSE_LOCATION, false);
```

Love the Optional use here, but the built-in methods on Map are probably more convenient...

Reply:
Nope, you were right!  Looking at this again, you _could_ get an NPE since the map might contain a `null` value and Java will throw an exception if you try and assign `null` to a primitive `boolean`.  This is good like you have it!